distributed/core.py

```diff
@@ -132,12 +132,12 @@ class Server(TCPServer):
     default_port = 0
 
     def __init__(self, handlers, max_buffer_size=MAX_BUFFER_SIZE,
-            connection_limit=512, **kwargs):
+            connection_limit=512, ssl_context_creator= create_ssl_context, **kwargs):
         self.handlers = assoc(handlers, 'identity', self.identity)
         self.id = str(uuid.uuid1())
         self._port = None
         self.rpc = ConnectionPool(limit=connection_limit)
-        super(Server, self).__init__(max_buffer_size=max_buffer_size, **kwargs)
+        super(Server, self).__init__(max_buffer_size=max_buffer_size, ssl_options=ssl_context_creator(), **kwargs)
```

Comment:
dependency injection for `ssl_context_creator` felt ok to me since it makes it clear that configurations are separated out. Perhaps, i haven't fully understood your concern regarding maintaining different API. Wouldn't `ssl_option=False` in `__init__` a deviation already?

I anticipate that `create_ssl_options` method to evolve into less trivial case and would like to keep it separate from the `Server` object.